### ISSUE TITLE
Symbol renamings are not taken into account

### ISSUE THREAD

**Issue by canonic-epicure** [TypeStrong/typedoc#1771]
## Search terms

Symbol, rename

## Steps to reproduce:
1) Clone the repo `https://github.com/canonic-epicure/typedoc-repro-1756`
2) Checkout the branch `re-export`
3) In repo directory: `npm i`
4) `npx typedoc index.ts index2.ts --validation.invalidLink`
5) Observe the output:
```
nickolay@frontier:~/workspace/typescript/typedoc-repro-1756$ npx typedoc index.ts index2.ts --validation.invalidLink
Warning: 
[MarkedLinksPlugin]: Found invalid symbol reference(s) in JSDocs, they will not render as links in the generated documentation.
  In index2: [[Test2.method]]
Info: Documentation generated at ./docs
```
6) Now if you comment line 1 in `index2.ts` and uncomment line 2:
```
// export { Test2 as Test, Test2 } from "./src/Test2.js"
export { Test2, Test2 as Test } from "./src/Test2.js"
```
and run the same command again, all works:
```
nickolay@frontier:~/workspace/typescript/typedoc-repro-1756$ npx typedoc index.ts index2.ts --validation.invalidLink
Info: Documentation generated at ./docs
```

The workaround is simple, however I think this case should be supported. Would it work if there would be `ts-doc` support?

**Comment by Gerrit0** [comment]
This is a surprisingly annoying bug to fix... because TypeDoc has two methods of resolving links, where links directly under the root project reflection get one method, and links within a class/namespace get a second, much more efficient method... I've been meaning to get rid of the special method for projects for a while now. This is as good enough of a reason as any.

However, this is a relatively large breaking change, since it does change how `{@link` and `[[` links are resolved, so it needs to wait for 0.23.

**Comment by canonic-epicure** [comment]
I see, ok, looking forward for 0.23.
